Valet/VALSecureEnclaveValet.h

```diff
@@ -21,19 +21,56 @@
 #import <Valet/VALValet.h>
 
 
-/// Reads and writes keychain elements that are stored on the Secure Enclave (supported on iOS 8.0 or later) using accessibility attribute VALAccessibilityWhenPasscodeSetThisDeviceOnly. Accessing or modifying these items will require the user to confirm their presence via Touch ID or passcode entry. If no passcode is set on the device, the below methods will fail. Data is removed from the Secure Enclave when the user removes a passcode from the device. Use the userPrompt methods to display custom text to the user in Apple's Touch ID and passcode entry UI.
-NS_CLASS_AVAILABLE_IOS(8_0)
+/// Compile flag for building against an SDK where Secure Enclave is available.
+#define VAL_SECURE_ENCLAVE_SDK_AVAILABLE ((TARGET_OS_IPHONE && __IPHONE_8_0) || (TARGET_OS_MAC && __MAC_10_11))
+
+/// Compile flag for building against an SDK where VALAccessControlTouchIDAnyFingerprint and VALAccessControlTouchIDCurrentFingerprintSet are available.
+#define VAL_ACCESS_CONTROL_TOUCH_ID_SDK_AVAILABLE (TARGET_OS_IPHONE && __IPHONE_9_0)
+
+/// Compile flag for building against an SDK where VALAccessControlDevicePasscode is available.
+#define VAL_ACCESS_CONTROL_DEVICE_PASSCODE_SDK_AVAILABLE ((TARGET_OS_IPHONE && __IPHONE_9_0) || (TARGET_OS_MAC && __MAC_10_11))
+
+
+#if VAL_SECURE_ENCLAVE_SDK_AVAILABLE
+
+typedef NS_ENUM(NSUInteger, VALAccessControl) {
+#if VAL_SECURE_ENCLAVE_SDK_AVAILABLE
```

Comment:
Just to socialize this a bit - after talking in person, we're trying to pull these compiler conditionals out of the header. Consumers really shouldn't have to parse this to use this class.